src/libopensc/card-dnie.c

```diff
@@ -940,32 +937,14 @@ static u8 *dnie_uncompress(sc_card_t * card, u8 * from, size_t *len)
 	if (uncompressed > MAX_FILE_SIZE)
 		goto compress_exit;
 
-	sc_log(card->ctx, "Data seems to be compressed. calling uncompress");
-	/* ok: data seems to be compressed */
-	upt = calloc(uncompressed, sizeof(u8));
-	if (!upt) {
-		sc_log(card->ctx, "alloc() for uncompressed buffer failed");
-		return NULL;
-	}
-	*len = uncompressed;
-	res = sc_decompress(upt,	/* try to uncompress by calling sc_xx routine */
-			    len,
-			    from + 8, (size_t) compressed, COMPRESSION_ZLIB);
```

Comment:
The buffer to decompress will start at offset 8. The first 8 bytes, however, will still be returned with dnie_read_binary. As consequence, this should give a decompression error.

If the file is decompressed, I think, the first 8 bytes need to be removed from the output buffer.

Reply:
I think this is now handled in `dnie_fill_cache()`.